google/cloud/aiplatform/tensorboard/uploader_utils.py

```diff
@@ -0,0 +1,297 @@
+          api: TensorboardServiceStub
+        """
+        self._experiment_resource_name = experiment_resource_name
+        self._api = api
+        self._run_name_to_run_resource_name: Dict[str, str] = {}
+        self._run_tag_name_to_time_series_name: Dict[(str, str), str] = {}
+
+    def get_run_resource_name(self, run_name: str):
+        """
+        Get the resource name of the run if it exists, otherwise creates the run
+        on One Platform before returning its resource name.
+        :param run_name: name of the run
+        :return: resource name of the run
+        """
+        if run_name not in self._run_name_to_run_resource_name:
+            tb_run = self._create_or_get_run_resource(run_name)
+            self._run_name_to_run_resource_name[run_name] = tb_run.name
+        return self._run_name_to_run_resource_name[run_name]
+
+    def _create_or_get_run_resource(self, run_name: str):
+        """Creates a new Run Resource in current Tensorboard Experiment resource.
+        Args:
+          run_name: The display name of this run.
+        """
+        tb_run = tensorboard_run.TensorboardRun()
+        tb_run.display_name = run_name
+        try:
+            tb_run = self._api.create_tensorboard_run(
+                parent=self._experiment_resource_name,
+                tensorboard_run=tb_run,
+                tensorboard_run_id=str(uuid.uuid4()),
+            )
+        except exceptions.InvalidArgument as e:
+            # If the run name already exists then retrieve it
+            if "already exist" in e.message:
+                runs_pages = self._api.list_tensorboard_runs(
+                    parent=self._experiment_resource_name
+                )
+                for tb_run in runs_pages:
+                    if tb_run.display_name == run_name:
+                        break
+
+                if tb_run.display_name != run_name:
+                    raise ExistingResourceNotFoundError(
+                        "Run with name %s already exists but is not resource list."
+                        % run_name
+                    )
+            else:
+                raise
+        return tb_run
+
+    def get_time_series_resource_name(
+        self,
+        run_name: str,
+        tag_name: str,
+        time_series_resource_creator: Callable[
+            [], tensorboard_time_series.TensorboardTimeSeries
+        ],
+    ):
+        """
+        Get the resource name of the time series corresponding to the tag, if it
+        exists, otherwise creates the time series on One Platform before
+        returning its resource name.
+        :param run_name: name of the run
+        :param tag_name: name of the tag
+        :param time_series_resource_creator: a constructor used for creating the
+        time series on One Platform.
+        :return: resource name of the time series
+        """
+        if (run_name, tag_name) not in self._run_tag_name_to_time_series_name:
```

Comment:
```suggestion
        if run_name, tag_name not in self._run_tag_name_to_time_series_name:
```

Reply:
Not sure if this syntax would work for dictionary lookup.

Reply:
You are right, leave it as is.